src/material/chips/_chips-theme.scss

```diff
@@ -1,101 +1,71 @@
-@use 'sass:color';
 @use 'sass:map';
-@use 'sass:meta';
 @use '@material/chips/chip-theme' as mdc-chip-theme;
-@use '@material/chips/chip-set' as mdc-chip-set;
-@use '@material/theme/theme-color' as mdc-theme-color;
-@use '@material/theme/color-palette' as mdc-color-palette;
-@use '@material/typography' as mdc-typography;
-@use '../core/mdc-helpers/mdc-helpers';
+@use '../core/tokens/m2/mdc/chip' as tokens-mdc-chip;
 @use '../core/theming/theming';
 @use '../core/typography/typography';
 
-@mixin _chip-variant($background, $foreground) {
-  @include mdc-chip-theme.theme((
-    elevated-container-color: $background,
-    elevated-disabled-container-color: $background,
-    label-text-color: $foreground,
-    disabled-label-text-color: $foreground,
-    with-icon-icon-color: $foreground,
-    with-icon-disabled-icon-color: $foreground,
-    with-trailing-icon-disabled-trailing-icon-color: $foreground,
-    with-trailing-icon-trailing-icon-color: $foreground,
-    with-icon-selected-icon-color: $foreground,
-  ));
-}
-
-@mixin _colored-chip($palette) {
+@mixin _colored-chip($config, $palette-name) {
+  $palette: map.get($config, $palette-name);
   $background: theming.get-color-from-palette($palette);
   $foreground: theming.get-color-from-palette($palette, default-contrast);
 
   &.mat-mdc-chip-selected,
   &.mat-mdc-chip-highlighted {
-    @include _chip-variant($background, $foreground);
+    @include mdc-chip-theme.theme((
+      elevated-container-color: $background,
```

Comment:
I think that the color variants are supposed to be generated in the tokens as well. It came up during the checkbox code review https://github.com/angular/components/pull/26744#discussion_r1130720783